lib/api/redfish-1.0/chassis.js

```diff
@@ -0,0 +1,232 @@
+                                'No Catalogs Found for Source (' + dataName + ').'
+                            );
+                        }
+                        return catalogs;
+                    });
+            });
+        case 'system':
+            return waterline.nodes.needByIdentifier(identifier)
+            .then(function(node) {
+                return _.filter(node.relations, function(relation) {
+                    return relation.relationType === 'encloses';
+                }).map(function(relation) {
+                    return relation.targets[0];
+                });
+            });
+        case 'chassis':
+            return waterline.nodes.needByIdentifier(identifier)
+            .then(function(node) {
+                return _.filter(node.relations, function(relation) {
+                    return relation.relationType === 'encloses';
+                }).map(function(relation) {
+                    return relation.targets;
+                });
+            })
+            .then(function(targets) {
+                return targets[0];
+            });
+        case 'chassisData':
+            return waterline.workitems.findPollers({node: identifier})
+            .filter(function(poller) {
+                return poller.config.command === 'chassis';
+            }).spread(function(poller) {
+                return taskProtocol.requestPollerCache(poller.id, { latestOnly: true });
+            }).then(function(data) {
+                var chassisData = data[0].chassis;
+                chassisData.power = (chassisData.power) ? 'On' : 'Off';
+                return chassisData;
+            });
+        case 'sdrData':
+            return waterline.workitems.findPollers({node: identifier})
+            .filter(function(poller) {
+                return poller.config.command === 'sdr';
+            }).spread(function(poller) {
+                return taskProtocol.requestPollerCache(poller.id, { latestOnly: true });
+            }).then(function(data) {
+                return data[0].sdr;
+            });
+    }
+};
+
+function listChassis(req, res) {
+    var options = {};
+    options.basepath = basepath;
+    options.templateScope = ['global'];
+    options.url = req.url;
+
+    waterline.nodes.find({type: 'enclosure'}).then(function(nodes) {
+        options.nodes = nodes;
+        return redfish.render('redfish.1.0.0.chassiscollection.json', 
+                            'ChassisCollection.json#/definitions/ChassisCollection',
+                            options);
+    }).then(function(output) {
+        res.status(200).json(output);
+    })
+    .catch(function(error) {
+        res.status(500).json(error);
+    });
+}
+
+function getChassis(req, res) {
+    var identifier = req.swagger.params.identifier.value;
+    var options = {};
+    options.basepath = basepath;
+    options.templateScope = ['global'];
+    options.url = req.url;
+    options.identifier = identifier;
+    options.chassisType = "Other";
```

Comment:
@jlongever this is such a boilerplate thing, maybe we should generate it from a factory instead?